### PR TITLE
Exercise: The sum of a range

### PR DESCRIPTION
**Pull request by artem-konovalov-personal**
Closed [#214](https://github.com/nesterone/sf-eng-init-gl-spring-2016/issues/214)

### CODE REVIEW

File: 01-syntax/03-data-structures/the-sum-of-ranges/range.js [hunk 1]
```diff
@@ -4,7 +4,29 @@
...
 
-// → Your code here.
+function range(start, end, step) {
+  var rangeStep;
+  var arr = [];
+  var arrIndex = 0;
+  var arrCounter;
+
+  if (start > end) {
+    rangeStep = step || -1;
```
Comment (nesterone) [inline 01-syntax/03-data-structures/the-sum-of-ranges/range.js]: code duplication ;-)

File: 01-syntax/03-data-structures/the-sum-of-ranges/range.js [hunk 2]
```diff
@@ -4,7 +4,29 @@
...
+
+    for (arrCounter = start; arrCounter >= end; arrCounter += rangeStep) {
+      arr[arrIndex] = arrCounter;
+      arrIndex++;
+    }
+  } else {
+    rangeStep = step || 1;
```
Comment (nesterone) [inline 01-syntax/03-data-structures/the-sum-of-ranges/range.js]: code duplication ;-)

**Comment by nesterone** [comment]
Try keep it DRY

**Comment by artem-konovalov-personal** [comment]
I have removed the code duplication from source code and also I have added one more check into the `range.js` file for the calculating of range with reduced numbers with a step by default (`range(10, 1)`)

**Pull request merged**